Lib/Src/CUartCom.cpp

```diff
@@ -98,8 +106,67 @@ void CUartCom::sendMessage(const std::string &msg)
     HAL_UART_Transmit(mp_huart, (uint8_t *)c_msg, msg_len, UART_TIMEOUT);
 
     // Disable UART_DE Pin
-    if (m_uart_de_port != nullptr)
+    m_uart_de_pin.set(GPIO_PIN_RESET);
+}
+
+/**
+ * @brief UART Reception Complete Interrupt Handler
+ * @arg p_huart Pointer to UART Handler
+ */
+void CUartCom::uartRxHandler(UART_HandleTypeDef *p_huart)
+{
+    // Store incoming char
+    if (m_rx_buffer.put((const char)m_rx_char))
+    {
+        // If end of string or rx_buffer is full, add to queue
+        std::string command_string = m_rx_buffer.get();
+
+        // If the queue has reached its max size, the message will be
+        // dismissed
+        if (m_cmd_queue.size() <= MAX_QUEUE_SIZE)
+        {
+            m_cmd_queue.push(command_string);
+        }
+    }
+
+    HAL_UART_Receive_IT(p_huart, &m_rx_char, 1);
+}
+
+/**
+ * @brief Check if the queue has any commands to read
+ * @return True if the queue is not empty
+ */
+bool CUartCom::isCommandAvailable()
+{
+    if (!m_cmd_queue.empty())
+    {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief Gets and deletes the first command in the queue
+ * @return Command as string
+ */
+std::string CUartCom::getCommand()
+{
+    std::string command = m_cmd_queue.front();
```

Comment:
Perhaps worth leaving a note here that `std::string` is returned by value, i.e. it is copied, so even when local copy gets out of scope the value is not destroyed.